cxx-sensors/src/main/java/org/sonar/cxx/sensors/coverage/TestwellCtcParser.java

```diff
@@ -0,0 +1,256 @@
+import java.util.regex.Matcher;
+
+import java.math.BigDecimal;
+
+import org.apache.commons.io.FilenameUtils;
+
+import org.sonar.api.batch.sensor.SensorContext;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+
+import static org.sonar.cxx.sensors.coverage.TestwellCtcResult.FILE_HEADER;
+import static org.sonar.cxx.sensors.coverage.TestwellCtcResult.FILE_RESULT;
+import static org.sonar.cxx.sensors.coverage.TestwellCtcResult.LINE_RESULT;
+import static org.sonar.cxx.sensors.coverage.TestwellCtcResult.SECTION_SEP;
+import static org.sonar.cxx.sensors.coverage.TestwellCtcResult.REPORT_FOOTER;
+
+
+
+/**
+ * {@inheritDoc}
+ */
+public class TestwellCtcParser extends CxxCoverageParser {
+
+  private static final Logger LOG = Loggers.get(TestwellCtcParser.class);
+  
+  private Scanner scanner;
+  private Matcher matcher;
+  private State state;
+  
+  
+  private static final int FROM_START = 0;
+  private static final int CONDS_FALSE = 1;
+  private static final int CONDS_TRUE = 2;
+  private static final int LINE_NR_GROUP = 3;
+  
+  private enum State {
+    BEGIN,
+    PARSING,
+    END;
+  }
+  
+
+  public TestwellCtcParser() {
+    // no operation but necessary for list of coverage parsers 
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public void processReport(final SensorContext context, File report, final Map<String, CoverageMeasures> coverageData) {
+    LOG.debug("Parsing 'Testwell CTC++' format");
+    
+    try {
+      this.scanner = new Scanner(report).useDelimiter(SECTION_SEP);
+      this.matcher = FILE_HEADER.matcher("");
+      
+      this.state = State.BEGIN;
+      parseReportHead();
+
+      while(State.PARSING == this.state) {
+          parseUnit(coverageData);
+      }
+      
+    } catch (FileNotFoundException e) {
+      LOG.warn("TestwellCtcParser file not found '{}'", e.getMessage());
+    }
+
+  }
+  private void parseReportHead() {
+    try {
+      if (!matcher.reset(scanner.next()).find()) {
+        LOG.info("'Testwell CTC++' file section not found.");
+      } else {
+        this.state = State.PARSING;
+      }
+    } catch (NoSuchElementException e) {
+      LOG.debug("'Testwell CTC++' file section not found!");
+    }
+  }
+  
+  private void parseUnit(final Map<String, CoverageMeasures> coverageData) {
+    LOG.debug(matcher.toString());
+    
+    if (matcher.usePattern(FILE_HEADER).find(FROM_START)) {
+      parseFileUnit(coverageData);
+    } else if (matcher.usePattern(REPORT_FOOTER).find(FROM_START)) {
+      state = State.END;
+      scanner.close();
+    }
+  }
+  
+  private void parseFileUnit(final Map<String, CoverageMeasures> coverageData) {
+    LOG.debug("Parsing file section...");
+    
+    String normalFilename;
+    String filename;
+    File file;
+    
+    filename = matcher.group(1);
+    if((new File(filename)).isAbsolute() == false) {
+      normalFilename = FilenameUtils.normalize("./" + filename);
+    }
+    else {
```

Comment:
Style: Move this "else" on the same line that the previous closing curly brace (4 times in this file)

Please align with the current style.